### ISSUE TITLE
Kind selector not accepted in the function prefix

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#142]
The prefix of a function declaration can include an intrinsic type. This type can be associated with a kind-selector according to the standard. Currently, the following code will fail with the FORTRAN front-end. 

```fortran 
module mod1
contains
  subroutine sub1()
  end subroutine sub1

  real(kind=8) function func1(a)
    integer :: a
    func1 = 10.0_8
  end function func1
end module mod1
```

This code is fine with the current version
```fortran 
module mod1
contains
  subroutine sub1()
  end subroutine sub1

  real(8) function func1(a)
    integer :: a
    func1 = 10.0_8
  end function func1
end module mod1
```

The following code does not work as well: 
```fortran
module mod1
contains
  subroutine sub1()
  end subroutine sub1

  real(8) elemental function func1(a)
    integer, intent(in) :: a
    func1 = 10.0_8
  end function func1
end module mod1
```

```bash
"funcprefix.f90:mod1", line 6: only function/subroutine statement are allowed in contains top level
"funcprefix.f90:mod1", line 6: syntax error
"funcprefix.f90:mod1", line 7: only function/subroutine statement are allowed in contains top level
"funcprefix.f90:mod1", line 8: only function/subroutine statement are allowed in contains top level
"funcprefix.f90:mod1", line 11: declaration among executables
```

**Comment by clementval** [comment]
I'll try to fix this myself :-)

**Comment by clementval** [comment]
The first or second rules are not taken with using the KIND keyword.
```
| type_spec KW FUNCTION IDENTIFIER dummy_arg_list KW result_opt bind_opt
{ $$ = list6(F_FUNCTION_STATEMENT, $4, $5, $1, NULL, $7, $8); }
| type_spec KW func_prefix FUNCTION IDENTIFIER dummy_arg_list KW result_opt bind_opt
{ $$ = list6(F_FUNCTION_STATEMENT, $5, $6, $1, $3, $8, $9); }
| func_prefix type_spec KW FUNCTION IDENTIFIER dummy_arg_list KW result_opt bind_opt
{ $$ = list6(F_FUNCTION_STATEMENT, $5, $6, $2, $1, $8, $9); }
```

**Comment by clementval** [comment]
Should be fixed by PR #149
